bin/title_case.py

```diff
@@ -0,0 +1,98 @@
+"""author_case.py
+
+usage: author_case.py -o <outdir>
+
+Try to correct titles that are written in mostly uppercase.
+"""
+
+import sys
+import os.path
+import glob
+import lxml.etree as etree
+import argparse
+import logging
+import re
+from fixedcase.common import get_text
+
+
+def titlecase(s):
+    if s is None:
+        return None
+    ret = []
+    first = True
+    for word in re.split(r"([^A-Za-z'’])", s):
+        if (
+            first
+            or word.lower()
+            not in "and or nor but a an the as at by for in of on per to".split()
+        ):
+            if len(word) > 0:
+                word = word[0].upper() + word[1:].lower()
+        else:
+            word = word.lower()
+        ret.append(word)
+        if any(c.isalpha() for c in word):
+            first = False
+        if word in [':', '(']:
+            first = True
+
+    ret = ''.join(ret)
+    return ret
+
+
+print(titlecase("A, B, AND C"))
+
```

Comment:
This should be removed or put in a `test` function?

Reply:
sorry